dali/kernels/common/utils.h

```diff
@@ -20,6 +20,16 @@
 namespace dali {
 namespace kernels {
 
+template <typename T, typename U>
+T Permute(const T& in, const U& permutation) {
+  T out = in;
```

Comment:
Some unit tests, kindly please ;)
Also, I don't like the fact, that you need to know, what to pass there. i.e. `permutation` has to have `.size()` and `operator[]` and `in` has to have `operator[]`. Can it be somehow handled in some meaningful compile-time error message?

Reply:
We decided not to complicate this with traits.